components/contribution-flow/utils.js

```diff
@@ -34,12 +34,11 @@ export const generatePaymentMethodOptions = (paymentMethods, stepProfile, stepDe
 
   let uniquePMs = uniqBy(paymentMethodsOptions, 'id');
 
-  // if ORG filter out 'collective' type payment
-  if (stepProfile.type === CollectiveType.ORGANIZATION) {
-    uniquePMs = uniquePMs.filter(
-      ({ paymentMethod }) => paymentMethod.providerType !== GQLV2_PAYMENT_METHOD_TYPES.ACCOUNT_BALANCE,
-    );
-  }
+  uniquePMs = uniquePMs.filter(
+    ({ paymentMethod }) =>
+      paymentMethod.providerType !== GQLV2_PAYMENT_METHOD_TYPES.ACCOUNT_BALANCE ||
+      collective.host?.legacyId === stepProfile.host?.id,
```

Comment:
In the contribution flow, we can normally assume that `host` always exists (https://github.com/opencollective/opencollective-frontend/blob/fix/edit-collective-sections-drop/pages/contribution-flow.js#L209).

```suggestion
      collective.host.legacyId === stepProfile.host?.id,
```